src/MSBuildLocator/DotNetSdkLocationHelper.cs

```diff
@@ -112,7 +123,33 @@ private static string GetDotNetBasePath(string workingDirectory)
                 return null;
             }
 
-            return matched.Groups[1].Value.Trim();
+            var lineSdkIndex = lines.FindIndex(line => line.Contains(".NET Core SDKs installed") || line.Contains(".NET SDKs installed"));
+
+            if (lineSdkIndex != -1)
+            {
+                lineSdkIndex++; 
+
+                while (lineSdkIndex < lines.Count && (!lines[lineSdkIndex].Contains(".NET Core runtimes installed") && !lines[lineSdkIndex].Contains(".NET runtimes installed")))
+                {
+                    var sdkMatch = SdkRegex.Match(lines[lineSdkIndex]);
+
+                    if (!sdkMatch.Success)
+                        break;
+
+                    var version = sdkMatch.Groups[1].Value.Trim();                     
+                    var path = sdkMatch.Groups[2].Value.Trim();
+
+                    path = Path.Combine(path, version) + "\\";                    
+
+                    // We insert at index 0 so that instance list will be sorted descending so that instances.FirstOrDefault() 
+                    // will always return the latest installed version of dotnet SDK 
+                    basePaths.Insert(0, path);      
```

Comment:
I don't see `dotnet` returning SDKs in a particular order documented anywhere. Also, this is O(N^2) and although not a big deal given the tiny number of items, it would be nicer to avoid it. My suggestion would be to return in the order as printed by `dotnet`, i.e. no sorting.

Reply:
You're correct though that if we don't add any smarts to this PR to pick the "right" SDK, we should keep the current behavior of returning the newest first. Maybe use `List<T>.Reverse()` then?

Reply:
For what it's worth, I don't think it's guaranteed that a foreach loop goes in order either. I don't think they'll ever change that unless they have aggressive parallelism, but it could be an argument for either explicitly sorting them or iterating (with a normal for loop) from top to bottom.

Reply:
The code now uses --list-sdks with a foreach loop and .Reverse. Need for any change to that?